rethinkdb/datadog_checks/rethinkdb/queries.py

```diff
@@ -0,0 +1,234 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Iterator, Mapping, Tuple
+
+import rethinkdb
+
+from .connections import Connection, ConnectionServer
+from .types import (
+    ClusterStats,
+    ConfigTotals,
+    CurrentIssuesTotals,
+    Job,
+    JoinRow,
+    ReplicaStats,
+    Server,
+    ServerStats,
+    ServerStatus,
+    ShardReplica,
+    Table,
+    TableStats,
+    TableStatus,
+)
+
+
+class QueryEngine(object):
+    """
+    Definition of RethinkDB queries used by the RethinkDB check.
+
+    Python ReQL reference documentation: https://rethinkdb.com/api/python/
+    """
+
+    def __init__(self, r=None):
+        # type: (rethinkdb.RethinkDB) -> None
+        self._r = rethinkdb.r if r is None else r
```

Comment:
Can we use a more clear name?

Reply:
What else would you suggest? I used `r` to fit the standard usage advertised across the RethinkDB docs, eg see [Run some queries](https://rethinkdb.com/docs/quickstart/#run-some-queries). Using something else would make the queries in `QueryEngine` harder to match with the RethinkDB docs (e.g. on selections, aggregation, joins, etc), thus harder to maintain IMO.

Reply:
runner? executor?

Reply:
Sorry, but I'm going to keep arguing in favor of keeping `r` (I added a comment to tell the reader about this choice). Eg neither notions of "runner" nor "executor" exist in the ReQL/RethinkDB jargon — the entrypoint to building ReQL queries is called `r` across the entire Internet — so that would be like calling a Coke "the fizzy dark brown beverage" when it's just known as Coke in the rest of the world. :-)